mongo/description/server.go

```diff
@@ -22,7 +22,8 @@ import (
 // This type should be used to track the state of a server that was selected to perform an operation.
 type SelectedServer struct {
 	Server
-	Kind TopologyKind
+	Kind    TopologyKind
+	HelloOK bool
```

Comment:
I figured this was the best place to store `HelloOK`. Let me know if you think the wrapped `Server` would be a better place for this field.

Reply:
I need to read a little more about how `description.SelectedServer` / `topology.SelectedServer` are used and interact with `description.Server` / `topology.Server` to give a definitive recommendation, but here are my initial thoughts.

I think the state of a server supporting the "hello" command is tied to the latest server description.

If a server responds with `helloOk: true`, it is guaranteed to support `hello` as long as the server is running. You cannot configure a server to stop supporting `helloOk` at runtime.

Not that C's server description is an ideal model, but `helloOk` is stored as part of the C server description in: https://github.com/mongodb/mongo-c-driver/blob/fdc2de4ff0e472a03c6c8a22dccaa1916ef3fd08/src/libmongoc/src/mongoc/mongoc-server-description.c#L605-L608

Given that, does it seem preferable to parse `helloOk` in `description.Server`?

Reply:
That makes sense to me, especially given there's precedent in the C driver. I've moved this field to the `Server` struct and now do the `helloOk` parsing in `description.NewServer()`.